docs/docker-image-build.md

```diff
@@ -0,0 +1,20 @@
+# Build a `ks` docker image
+
+Copy and paste the following commands:
+```bash
+# Clone the ksonnet repo into your GOPATH
+go get github.com/ksonnet/ksonnet
+
+# Build and install binary under shortname `ks` into $GOPATH/bin
+cd $GOPATH/src/github.com/ksonnet/ksonnet
+dep ensure
+make docker-image
+```
+
+# Running `ks` via docker
+
+In order to run via docker, the `ks` process needs access to a kubernetes config file, certificates, and the current working directory. We set these up using bind mounts:
+
+```bash
+docker run -e KUBECONFIG="${KUBECONFIG}" --mount type=bind,source="${KUBECONFIG}",target="${KUBECONFIG}" --mount type=bind,source="${HOME}/.minikube",target="${HOME}/.minikube" --mount type=bind,source="$(pwd)",target="$(pwd)" -w "$(pwd)" ks --help
```

Comment:
This fails for people like me who lack a `KUBECONFIG` variable.
Let's supply a default:

`${KUBECONFIG}` -> `${KUBECONFIG:-${HOME}/.kube/config}`

Reply:
The minikube dependency is not always going to be correct - perhaps what we really give is a bash function that parses out the certificate paths from `${KUBECONFIG}` and builds the arguments that way? Or is that overkill?

Reply:
oh man i know - i actually had the exact line you suggest for `KUBECONFIG` with default ... and i had a CERTDIR with default to ${HOME}/.kube ... and i talked to @bryanl about parsing the cert file but apparently there are versions of the cert specification that are links on which actions need to be taken so ... that's exciting.

the ergonomics of this are why i didn't include something to add to `.bash_aliases` - it's way to user specific to do something simple. we can make this work intuitively but it might have diminishing returns.

Reply:
Agreed. Maybe we attack this with documentation, describing what needs to be mounted logically and giving an example.

Reply:
lol - yeah - this is the docs that i meant that to be - i'll add a couple other examples and note failure cases as well to help people navigate.